core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/base/BaseAuthorizationRule.scala

```diff
@@ -33,7 +33,7 @@ import tech.beshu.ror.accesscontrol.domain.{Group, LoggedUser, User}
 import tech.beshu.ror.utils.uniquelist.{UniqueList, UniqueNonEmptyList}
 
 trait BaseAuthorizationRule extends AuthorizationRule with SimpleAuthorizationImpersonationSupport {
-
+  protected def calculateAllowedGroupsForUser(usersGroups: UniqueNonEmptyList[Group]): Option[UniqueNonEmptyList[Group]]
```

Comment:
in the `BaseGroupsRule` class we have a similar method:

```
protected def availableGroupsFrom(userGroups: Set[Group], ruleGroups: Set[Group]): Option[UniqueNonEmptyList[Group]]```

maybe it'd be nice to do it here in the same way?
1. it looks it's easier to reason about the method when it has two parameters
2. do you think the current name is better than the one from BaseGroupsRule? if so, maybe we should modify the BaseGroupsRule one? If not, let's make this one similar to the one from BaseGroupsRule
3. It'd ve nice to add `\n`s before and after the method declaration

Reply:
what about point 1?